src/plugins/persistence/couch/CouchChangesFeed.js

```diff
@@ -61,10 +68,34 @@
 
         couchEventSource = new EventSource(url);
         couchEventSource.onerror = self.onerror;
+        couchEventSource.onopen = self.onopen;
 
         // start listening for events
         couchEventSource.addEventListener('message', self.onCouchMessage);
         connected = true;
         console.debug('⇿ Opened connection ⇿');
     };
+
+    self.updateCouchStateIndicator = function () {
+        const { readyState } = couchEventSource;
+        let message = {
+            type: 'state',
+            state: 'pending'
+        };
+        switch (readyState) {
+        case EventSource.CONNECTING:
+            message.state = 'pending';
+            break;
+        case EventSource.OPEN:
+            message.state = 'open';
+            break;
+        case EventSource.CLOSED:
+            message.state = 'close';
+            break;
+        }
```

Comment:
We should have a default case here - and maybe throw an error if we get something strange?